### PR TITLE
all posts api connected

### PR DESCRIPTION
**Pull request by sNkr-10**

### CODE REVIEW

File: services/PostService.js [hunk 1]
```diff
@@ -280,15 +280,29 @@ async function getPostsByQuery(query, clickNo) {
...
+      params: { username: username }
+    });
+    return data;
+  } catch (err) {
+    console.log(err);
+    return;
+  }
+}
+
+async function getAllPostsByUsername(username) {
```
Comment (thealpha93) [inline services/PostService.js:298]: Add proper comments on function definition

File: services/PostService.js [hunk 2]
```diff
@@ -280,15 +280,29 @@ async function getPostsByQuery(query, clickNo) {
...
+  let url = "";
+  if (typeof window == "undefined") url = serverUrl;
+  else url = baseUrl;
```
Comment (thealpha93) [inline services/PostService.js:301]: Add a line break and write comments on what this code does.  better to have write it as a single function that can be reused

File: services/PostService.js [hunk 3]
```diff
@@ -280,15 +280,29 @@ async function getPostsByQuery(query, clickNo) {
...
   if (typeof window == "undefined") url = serverUrl;
   else url = baseUrl;
   try {
     const { data } = await axios.get(`${url}/${allPostsUrl}/count`, {
-      params: { userId: userId }
+      params: { username: username }
+    });
+    return data;
+  } catch (err) {
+    console.log(err);
```
Comment (thealpha93) [inline services/PostService.js:293]: If you return nothing from the catch block you won't be able to handle the error properly in the component.

**Review by thealpha93 — CHANGES_REQUESTED** [review]

File: services/PostService.js (hunk 2 above)
Comment (sNkr-10) [reply services/PostService.js:301]: made it as a function getUrl located at lib/getUrl

File: services/PostService.js (hunk 3 above)
Comment (sNkr-10) [reply services/PostService.js:293]: return err.message

File: services/UserService.js [hunk 4]
```diff
@@ -24,14 +20,14 @@ async function getLatestUsers(reqParams) {
         return response;
     } catch (err) {
         console.log(err);
-        return;
+        return err.message;
```
Comment (thealpha93) [inline services/UserService.js:23]: Don't return err as such, use throw

**Review by thealpha93 — CHANGES_REQUESTED** [review]

**Pull request merged**